examples/hpc/dla-driver/src/lib.rs

```diff
@@ -86,35 +112,39 @@ pub struct LayerConfig {
 
 #[derive(Clone, Copy)]
 #[rustfmt::skip]
+/// Data banks in DLA's memory buffer, stores inputs, kernels and outputs.
 pub enum MemoryBank {
     Bank0, Bank1, Bank2, Bank3, Bank4, Bank5, Bank6, Bank7, Bank8, Bank9,
     Bank10, Bank11, Bank12, Bank13, Bank14, Bank15,
 }
 
-impl MemoryBank {
-    fn from_u32(value: u32) -> MemoryBank {
+impl TryFrom<u32> for MemoryBank {
+    type Error = &'static str;
```

Comment:
If you're using a string, at least make it informative :-)

I'd use a unit type (`()`) or a trivial tuple struct here (`struct InvalidBankIdx(u32);`), since it should be obvious at the call site what caused the error. In the unlikely case the runtime/driver implementor wants to add more info, they can impl Debug for this struct.

Reply:
I know from experience that `derive(Debug)` can easily become an issue on code size constrained implementations (such as anything on SysCtrl). What I'd do nowadays would be to add an optional `derive(Debug)` on `InvalidBankIdx`, either by using `cfg_attr(debug_assertions, derive(Debug))` or providing a feature such as `debug-traits`.

Reply:
Changed to use unit type.